File: trip-server-py/src/models/trip.py
"""Trip model (trips table)"""

from sqlalchemy import Column, Integer, String, ForeignKey, JSON
from sqlalchemy.orm import relationship

from src.models.base import Base, BaseModelWithoutUpdatedAt


class Trip(Base, BaseModelWithoutUpdatedAt):
    """Trip model matching Prisma schema (no updated_at field)"""
    
    __tablename__ = "trips"
    __table_args__ = (
        {"comment": "行程表"},
    )
    
    # Fields
    user_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=True,
        index=True,
        comment="用户ID（外键，可为空）"
    )
    from_city = Column(
        String(50),
        nullable=True,
        comment="出发城市"
    )
    city = Column(
        String(50),
        nullable=False,
        comment="目的地城市"
    )
    days = Column(
        Integer,
        nullable=False,
        comment="行程天数"
    )
    budget = Column(
        Integer,
        nullable=False,
        comment="预算"
    )
    content = Column(
        JSON,
        nullable=False,
        comment="行程内容（JSON）"
    )
    status = Column(
        String(20),
        default="completed",
        nullable=False,
        comment="状态"
    )
    parent_trip_id = Column(
        Integer,
        ForeignKey("trips.id"),
        nullable=True,
        comment="父行程ID（自引用）"
    )
    
    # Relationships
    user = relationship("User", back_populates="trips")
    parent = relationship(
        "Trip",
        remote_side="Trip.id",
        back_populates="versions",
        foreign_keys=[parent_trip_id]
    )
    versions = relationship(
        "Trip",
        back_populates="parent",
        foreign_keys=[parent_trip_id]
    )
    
    def __repr__(self):
        return f"<Trip(id={self.id}, city={self.city}, days={self.days})>"

File: trip-backend-java/src/main/java/com/trip/backend/domain/entity/Trip.java
package com.trip.backend.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;
import java.util.Map;

/**
 * 行程实体（对应 Python models/trip.py）
 * - content 为 JSONB 列
 * - parent_trip_id 自引用（版本链）
 * - 无 updated_at
 */
@Entity
@Table(name = "trips", indexes = {
    @Index(name = "idx_trips_user_id", columnList = "user_id"),
    @Index(name = "idx_trips_status", columnList = "status"),
    @Index(name = "idx_trips_parent_trip_id", columnList = "parent_trip_id")
})
@Getter
@Setter
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(length = 100)
    private String fromCity;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(nullable = false)
    private Integer days;

    @Column(nullable = false)
    private Integer budget;

    // JSONB 列
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private Map<String, Object> content;

    @Column(nullable = false, length = 50)
    private String status = "completed"; // candidate / completed / discarded

    // 版本链（自引用）
    @Column(name = "parent_trip_id")
    private Long parentTripId;

    @Column(nullable = false)
    private OffsetDateTime createdAt;

    // 无 updated_at

    protected Trip() {}

    @PrePersist
    protected void onCreate() {
        this.createdAt = OffsetDateTime.now();
    }
}
